tests/test_integration_layout.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+import asyncio
+import platform
+from typing import cast, List
+
+import pytest
+
+from tests.utilities.test_app import AppTest
+from textual.app import ComposeResult
+from textual.geometry import Size
+from textual.widget import Widget
+from textual.widgets import Placeholder
+
+PLATFORM = platform.system()
+WINDOWS = PLATFORM == "Windows"
```

Comment:
we have to remove 1 pixel to the expected widths when the test runs on Windows - which is a hard-coded behaviour in the Rich `Console` iirc